### PR TITLE
Move add_data_source from the dlme repo

### PR DESCRIPTION
**Pull request by jcoyne**
Ref sul-dlss/dlme#437

### CODE REVIEW

File: docs/add_data_source.md [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+3. [Add Your Data Source's Settings](#3-add-your-data-source-settings)
+4. [Establish data mapping](#4-create-your-data-mapping)
+5. [Write mapping configuration](#5-create-your-mapping-configuration-file)
+6. [Optional: Write mapping macro for data transformations](#6-mapping-macros)
+7. [Write spec to test data mapping and provide fixture data for testing](#7-test-your-data-mapping)
+8. [Create Your Pull Request](#8-create-a-pull-request)
+
+## 1. Stage Your Metadata
+
+At present, this DLME prototype loads metadata from this DLME Metadata GitHub repository: https://github.com/waynegraham/dlme-metadata. You will need to add your metadata - in the appropriate directory - via a pull request to that DLME Metadata GitHub repository for it to be loaded by our pipeline.
```
Comment (peetucket) [inline docs/add_data_source.md]: are we really using that repo, or the fork here?  https://github.com/sul-dlss/dlme-metadata/

File: docs/add_data_source.md [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+
+## 2. Create a DLME GitHub Repository Branch
+
+With your metadata staged in the [DLME Metadata GitHub repository](https://github.com/waynegraham/dlme-metadata), you now need prepare a branch - to then become a pull request when you're done - on this DLME application GitHub repository. This branch will add your new data source configuration, update the loading settings to load your new metadata, add a data conversion test for your new metadata mapping, and add any other optional code for your new data conversion.
```
Comment (peetucket) [inline docs/add_data_source.md]: should be https://github.com/sul-dlss/dlme-metadata/ ?

File: docs/add_data_source.md [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+
+## 3. Add Your Data Source Settings
+
+You need to update the `config/settings.yml` file for this application codebase to include information for the new data source. To do so:
+
+1. At the end of the sources section in [config/settings.yml](../config/settings.yml), following the existing format (see the example below).
+2. Give it a meaningful and unique data source name: `stanford_mods:` or `my_new_source:`.
+3. Indicate the directory within the [DLME metadata repository](https://github.com/waynegraham/dlme-metadata) where your metadata files are located.
```
Comment (peetucket) [inline docs/add_data_source.md]: https://github.com/sul-dlss/dlme-metadata/ ?

File: docs/add_data_source.md [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+
+At present, this DLME prototype loads metadata from this DLME Metadata GitHub repository: https://github.com/waynegraham/dlme-metadata. You will need to add your metadata - in the appropriate directory - via a pull request to that DLME Metadata GitHub repository for it to be loaded by our pipeline.
+
+## 2. Create a DLME GitHub Repository Branch
+
+With your metadata staged in the [DLME Metadata GitHub repository](https://github.com/waynegraham/dlme-metadata), you now need prepare a branch - to then become a pull request when you're done - on this DLME application GitHub repository. This branch will add your new data source configuration, update the loading settings to load your new metadata, add a data conversion test for your new metadata mapping, and add any other optional code for your new data conversion.
+
+## 3. Add Your Data Source Settings
+
+You need to update the `config/settings.yml` file for this application codebase to include information for the new data source. To do so:
```
Comment (peetucket) [inline docs/add_data_source.md]: i think this is different now (is it sul-dlss/dlme-transform/spec/fixtures/metadata_mappings.json)?

**Review by peetucket — COMMENTED** [review]
This may need some updates to reflect the new process (or just wholesale re-writing or removal) to avoid being in conflict with the new mechanism?

**Comment by jcoyne** [comment]
I don't know the answers to most of these. I just moved the file without edits at this point.

File: docs/add_data_source.md (hunk 4 above)
Comment (jcoyne) [reply docs/add_data_source.md]: No, that's just for testing.  It's here: https://github.com/sul-dlss/dlme-traject/blob/master/metadata_mapping.json

**Comment by peetucket** [comment]
OK, I can make some small tweaks based on my understand and put a caveat in there for the rest.  we can make further change in other PRs

File: docs/add_data_source.md [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+
+`rspec spec/import/my_new_source_spec.rb`
+
+This will produce a report that indicates any failures - where the conversion output and the spec's expectations do not match.
+
+If you want to run a full report of your branch's changes (this is optional), you can run the following to run rubocop (which checks your Ruby syntax), runs all the tests, and checks for code coverage (if tests exist for all the code in this database):
+
+`coveralls report`
+
+## 8. Create a Pull Request
```
Comment (jcoyne) [inline docs/add_data_source.md]: This section is completely out of date.

File: docs/add_data_source.md [hunk 6]
```diff
@@ -0,0 +1,357 @@
...
+
+Macros are additions to Traject that allow for more complication mappings - fields you need to normalize, add conditionals to, or other types of logic, beyond the availability of the current Traject DSL. Adding Macros is entirely optional and dependent on the needs of the particular data to be translated as well as your comfort with Ruby and lambdas.
+
+Traject macros provide a facility for more complex data transformation. For more information on how to use macros, you can [review the existing macros](../lib/traject/macros) for examples, or check out the original [Traject documentation on macros and custom logic](https://github.com/traject/traject/blob/master/doc/indexing_rules.md).
+
+## 7. Test Your Data Mapping
+
+Consult the README in the [dlme-transform repo](https://github.com/sul-dlss/dlme-transform) to see how to run traject configs locally.
+
+### Add Data Tests
```
Comment (jcoyne) [inline docs/add_data_source.md]: This section is completely out of date

File: docs/add_data_source.md [hunk 7]
```diff
@@ -0,0 +1,357 @@
...
+      expect(dlme['agg_provider']).to eq 'EXPECTED VALUE FROM FIXTURE'
+      ..repeat for all fields to test..
+    end
+  end
+end
+```
+
+You can see examples of relevant tests in that `spec/import` folder.
+
+### Run Data Tests Locally
```
Comment (jcoyne) [inline docs/add_data_source.md]: This section is completely out of date

File: docs/add_data_source.md [hunk 8]
```diff
@@ -285,66 +285,7 @@ Example:
...
-This will produce a report that indicates any failures - where the conversion output and the spec's expectations do not match.
-
-If you want to run a full report of your branch's changes (this is optional), you can run the following to run rubocop (which checks your Ruby syntax), runs all the tests, and checks for code coverage (if tests exist for all the code in this database):
-
-`coveralls report`
-
-## 8. Create a Pull Request
+## 6. Create a Pull Request
 
 After you have added your mapping configuration, updated the settings to be able to pull your metadata, added your metadata to the DLME Metadata repository, and done any testing needed, you want to create a pull request on this DLME repository to add your data to the master branch.
```
Comment (jcoyne) [inline docs/add_data_source.md]: I think this paragraph needs to go too.

Comment (jcoyne) [reply docs/add_data_source.md]: Or at least change "DLME repository" to "DLME Metadata repository"

**Pull request merged**